src/wallet/wallet.h

```diff
@@ -658,7 +658,7 @@ class CWallet final : public WalletStorage, public interfaces::Chain::Notificati
     CAmount GetDebit(const CTransaction& tx, const isminefilter& filter) const;
     void chainStateFlushed(const CBlockLocator& loc) override;
 
-    DBErrors LoadWallet();
+    DBErrors LoadWallet(bool& rescan_required);
```

Comment:
In a751305ed7d0e085d80c76dfe1b419858ce331df "Corrupt wallet tx shouldn't trigger rescan of all wallets"

It seems like it would be cleaner to add a `DBErrors:RESCAN_REQUIRED` instead of this `rescan_required` parameter and requiring a bunch of callers to provide a dummy bool.

Reply:
I like this approach too.

Reply:
That requires changing https://github.com/bitcoin/bitcoin/blob/419afa93419e6840f78cb94b4a39d826eb10e139/src/wallet/wallet.cpp#L2546

A low effort alternative is to use pointer:
```cpp
DBErrors LoadWallet(bool* rescan_required = nullptr);
```
or overload
```cpp
DBErrors LoadWallet() { bool dummy; return LoadWallet(dummy); }
```

Reply:
I'll overload LoadWallet as @promag suggests :+1:
I was just erring on the side of caution when it came to hiding the parameter, so it wasn't accidentally ignored.

Reply:
I really don't think it should be an out parameter since we have this enum for loading errors.

Reply:
Ok fair enough, I'll use your approach